packages/js/product-editor/src/hooks/use-product.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { Product, ReadOnlyProperties } from '@woocommerce/data';
```

Comment:
We can explicitly define its importing types

```suggestion
import type { Product, ReadOnlyProperties } from '@woocommerce/data';
```